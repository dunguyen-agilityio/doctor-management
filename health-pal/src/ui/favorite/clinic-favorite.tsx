import { Stack } from 'tamagui'

import { FAVORITE_EMPTY } from '@app/constants'

import { ClinicCard, LoadingIndicator } from '@app/components'
import Empty from '@app/components/empty'
import ErrorState from '@app/components/error'
import { useSession } from '@app/contexts'
import { useFavoriteHospitals } from '@app/hooks/use-favorite'
import { Clinic, TClinicFavorite } from '@app/models/clinic'
import { FAVORITE_TYPES } from '@app/types/favorite'

import HospitalList from '../hospital/hospital-list'

const ItemSeparatorComponent = () => <Stack height={12} />

const keyExtractor = (item: Clinic) => item.documentId

const ClinicFavorite = () => {
  const { session } = useSession()
  const { jwt, user } = session ?? {}
  const { data: hospitals, isLoading, error, refetch } = useFavoriteHospitals(user!.id, jwt!)

  if (isLoading) return <LoadingIndicator />

  if (!hospitals || error)
    return (
      <ErrorState
        title="Error Loading Favorites"
        message="We couldn't load your favorite Hospital. Please try again."
        onRetry={refetch}
      />
    )

  const renderItem = ({ item }: { item: TClinicFavorite }) => {
    return <ClinicCard px={24} h={256} full {...item} />
  }

  const ListEmptyComponent = <Empty {...FAVORITE_EMPTY[FAVORITE_TYPES.HOSPITAL]} />

  return (
    <HospitalList
      data={hospitals}
      ItemSeparatorComponent={ItemSeparatorComponent}
      keyExtractor={keyExtractor}
      renderItem={renderItem}
      ListEmptyComponent={ListEmptyComponent}
    />
  )
}

export default ClinicFavorite
