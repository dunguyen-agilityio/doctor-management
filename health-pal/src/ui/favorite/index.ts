export { default as DoctorFavorite } from './doctor-favorite'
export { default as FavoriteButton } from './favorite-button'
export { default as HospitalFavorite } from './hospital-favorite'
