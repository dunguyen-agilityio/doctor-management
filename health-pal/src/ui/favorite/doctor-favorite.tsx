import { FAVORITE_EMPTY } from '@app/constants'

import DoctorCard from '@app/components/doctor-card'
import DoctorList from '@app/components/doctor-list'
import Empty from '@app/components/empty'
import ErrorState from '@app/components/error'
import LoadingIndicator from '@app/components/loading-indicator'
import { useSession } from '@app/contexts'
import { useFavoriteDoctors } from '@app/hooks/use-favorite'
import { TDoctorData } from '@app/models/doctor'
import { FAVORITE_TYPES } from '@app/types/favorite'
import { formatDoctor } from '@app/utils/doctor'

const DoctorFavorite = () => {
  const { session } = useSession()

  const { jwt, user } = session ?? {}
  const {
    data: doctors,
    error: favoriteError,
    isLoading: favoriteLoading,
    refetch,
  } = useFavoriteDoctors(user!.id, jwt!)

  if (favoriteLoading) {
    return <LoadingIndicator />
  }

  if (!doctors || favoriteError) {
    return (
      <ErrorState
        title="Error Loading Favorites"
        message={`We couldn't load your favorite Doctor. Please try again.`}
        onRetry={refetch}
      />
    )
  }

  const renderItem = ({ item }: { item: TDoctorData }) => {
    return <DoctorCard {...formatDoctor(item)} />
  }

  const ListEmptyComponent = <Empty {...FAVORITE_EMPTY[FAVORITE_TYPES.DOCTOR]} />

  return (
    <DoctorList data={doctors} renderItem={renderItem} ListEmptyComponent={ListEmptyComponent} />
  )
}

export default DoctorFavorite
