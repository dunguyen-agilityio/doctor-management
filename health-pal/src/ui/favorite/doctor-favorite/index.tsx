import { FAVORITE_EMPTY } from '@app/constants'

import { useFavoriteDoctors } from '@app/hooks/use-favorite'

import { Empty, ErrorState } from '@app/components'
import DoctorListSkeleton from '@app/components/skeleton/doctor-list-skeleton'

import DoctorList from '@app/ui/doctor/doctor-list'

import { FAVORITE_TYPES } from '@app/types/favorite'

const DoctorFavorite = () => {
  const {
    data: doctors,
    error: favoriteError,
    isLoading: favoriteLoading,
    refetch,
  } = useFavoriteDoctors()

  if (favoriteLoading) {
    return <DoctorListSkeleton />
  }

  if (!doctors || favoriteError) {
    return (
      <ErrorState
        title="Error Loading Favorites"
        message={`We couldn't load your favorite Doctor. Please try again.`}
        onRetry={refetch}
      />
    )
  }

  const ListEmptyComponent = <Empty {...FAVORITE_EMPTY[FAVORITE_TYPES.DOCTOR]} />

  return <DoctorList data={doctors} ListEmptyComponent={ListEmptyComponent} />
}

export default DoctorFavorite
