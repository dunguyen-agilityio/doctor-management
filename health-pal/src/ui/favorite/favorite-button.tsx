import React from 'react'

import { ButtonProps } from 'tamagui'

import { Button } from '@theme'

import HeartFill from '@icons/heart-fill'
import HeartOutline from '@icons/heart-outline'

import { useSession } from '@app/contexts'
import { QUERY_KEY, queryClient } from '@app/react-query.config'
import { addFavorite, deleteFavorite } from '@app/services/favorite'
import { FAVORITE_TYPES } from '@app/types/favorite'

import { tokens } from '@/tamagui.config'

const VARIANT_STYLE = {
  primary: {
    backgroundColor: 'rgba(31, 42, 55, 0.20)',
    padding: 6,
    pressStyle: {
      backgroundColor: 'rgba(31, 42, 55, 0.20)',
    },
  },
  secondary: {
    backgroundColor: 'transparent',
  },
} satisfies Record<string, ButtonProps>

interface FavoriteButtonProps {
  variant?: keyof typeof VARIANT_STYLE
  color?: string
  favoriteId?: string
  type: FAVORITE_TYPES
  data: string
}

const FavoriteButton = ({
  favoriteId,
  variant = 'primary',
  color = tokens.color.primary.val,
  type,
  data,
}: FavoriteButtonProps) => {
  const { session } = useSession()
  const renderIcon = () => {
    if (favoriteId) return <HeartFill fill={color} width={15} height={15} />

    return <HeartOutline stroke={color} width={15} height={15} />
  }

  const { jwt, user } = session ?? {}
  const userId = user?.id

  const { CLINIC_FAVORITE, DOCTOR_FAVORITE } = QUERY_KEY
  const queryKey = type === FAVORITE_TYPES.DOCTOR ? DOCTOR_FAVORITE : CLINIC_FAVORITE

  const handleRemove = async () => {
    if (!favoriteId || !jwt) return

    try {
      await deleteFavorite(favoriteId, jwt)

      await queryClient.invalidateQueries({ queryKey })
    } catch (error) {
      console.log('error', error)
    }
  }

  const handleAdd = async () => {
    if (!jwt || !userId) return

    try {
      await addFavorite(type, data, jwt, userId)
      await queryClient.invalidateQueries({ queryKey })
    } catch (error) {
      console.log('error', error)
    }
  }

  return (
    <Button
      {...VARIANT_STYLE[variant]}
      onPress={favoriteId ? handleRemove : handleAdd}
      position="absolute"
      right={6}
      top={6}
      zIndex={1000}
      width={15}
      height={15}
      variant="icon">
      {renderIcon()}
    </Button>
  )
}

export default FavoriteButton
