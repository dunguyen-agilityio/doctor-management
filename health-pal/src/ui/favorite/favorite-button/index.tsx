import { Button, ButtonProps } from '@theme'

import HeartFill from '@icons/heart-fill'
import HeartOutline from '@icons/heart-outline'

import { useAddFavorite } from '@app/hooks/use-add-favorite'
import { useRemoveFavorite } from '@app/hooks/use-remove-favorite'
import { FAVORITE_TYPES } from '@app/types/favorite'

import { tokens } from '@/tamagui.config'

const VARIANT_STYLE = {
  [FAVORITE_TYPES.HOSPITAL]: {
    backgroundColor: 'rgba(31, 42, 55, 0.20)',
    padding: 6,
    pressStyle: {
      backgroundColor: 'rgba(31, 42, 55, 0.20)',
    },
  },
  [FAVORITE_TYPES.DOCTOR]: {
    backgroundColor: 'transparent',
  },
} satisfies Record<string, ButtonProps>

const COLOR = {
  [FAVORITE_TYPES.DOCTOR]: tokens.color.primary.val,
  [FAVORITE_TYPES.HOSPITAL]: tokens.color.white.val,
}

type FavoriteButtonProps = {
  color?: string
  size?: number
  type?: FAVORITE_TYPES
  itemId: number
  itemName: string
  favoriteId?: string
} & ButtonProps

const FavoriteButton = ({
  type = FAVORITE_TYPES.DOCTOR,
  color = COLOR[type],
  size = 15,
  itemId,
  itemName,
  favoriteId,
  ...props
}: FavoriteButtonProps) => {
  const { mutate: removeFavorite, isPending: removeFavPending } = useRemoveFavorite(type, itemName)

  const { mutate: addFavorite, isPending: addFavPending } = useAddFavorite(type, itemName)

  const handleFavorite = () => {
    if (favoriteId) {
      removeFavorite(favoriteId)
      return
    }

    addFavorite(itemId)
  }

  const disabled = !!removeFavPending || !!addFavPending

  const renderIcon = () => {
    if (favoriteId) return <HeartFill fill={color} width={size} height={size} />

    return <HeartOutline stroke={color} width={size} height={size} />
  }

  return (
    <Button
      testID="favorite-button"
      {...VARIANT_STYLE[type]}
      width={size}
      height={size}
      variant="icon"
      zIndex={100}
      disabled={disabled}
      disabledStyle={{ opacity: 0.8, backgroundColor: '$colorHover' }}
      onPress={handleFavorite}
      {...props}>
      {renderIcon()}
    </Button>
  )
}

export default FavoriteButton
