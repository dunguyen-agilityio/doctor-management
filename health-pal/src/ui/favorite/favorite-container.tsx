import { Stack } from 'tamagui'

import { useAddFavorite } from '@app/hooks/use-add-favorite'
import { useRemoveFavorite } from '@app/hooks/use-remove-favorite'
import { FAVORITE_TYPES } from '@app/types/favorite'

import { tokens } from '@/tamagui.config'

import FavoriteButton from './favorite-button'

interface FavoriteContainerProps {
  favoriteId?: string
  type: FAVORITE_TYPES
  itemId: number
  itemDocId: string
  itemName: string
}

const COLORS = {
  [FAVORITE_TYPES.DOCTOR]: tokens.color.primary.val,
  [FAVORITE_TYPES.HOSPITAL]: tokens.color.white.val,
}

const FavoriteContainer = ({
  children,
  ...props
}: React.PropsWithChildren<FavoriteContainerProps>) => {
  const { favoriteId, type, itemId, itemName } = props

  const { mutate: removeFavorite, isPending: removeFavPending } = useRemoveFavorite(type, itemName)

  const { mutate: addFavorite, isPending: addFavPending } = useAddFavorite(type, itemName)

  const handleFavorite = () => {
    if (favoriteId) {
      removeFavorite(favoriteId)
      return
    }

    addFavorite(itemId)
  }

  const disabled = !!removeFavPending || !!addFavPending

  return (
    <Stack position="relative">
      <Stack
        position="absolute"
        borderRadius={12}
        inset={0}
        opacity={disabled ? 0.85 : 1}
        backgroundColor="$shadow1"
        zIndex={disabled ? 1000 : 0}
      />
      <FavoriteButton
        color={COLORS[type]}
        position="absolute"
        top={6}
        right={6}
        zIndex={100}
        onPress={handleFavorite}
        {...props}
      />
      {children}
    </Stack>
  )
}

export default FavoriteContainer
