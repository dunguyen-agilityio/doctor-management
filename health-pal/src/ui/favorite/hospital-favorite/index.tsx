import { keyExtractor } from '@/utils/list'
import { getMediaQuery } from '@/utils/media-query'

import { StyleSheet } from 'react-native'

import { Stack } from 'tamagui'

import { FAVORITE_EMPTY } from '@/constants'

import { useFavoriteHospitals } from '@/hooks/use-favorite'

import { Empty, ErrorState } from '@/components'
import HospitalListSkeleton from '@/components/skeleton/hospital-list-skeleton'

import HospitalList from '@/ui/hospital/hospital-list'

import { FAVORITE_TYPES } from '@/types/favorite'

const ItemSeparatorComponent = () => <Stack height={12} />

const HospitalFavorite = () => {
  const { height } = getMediaQuery({ px: 24, height: 256, full: true })
  const { data: hospitals, isLoading, error, refetch } = useFavoriteHospitals()

  if ((!isLoading && !hospitals) || error) {
    return (
      <ErrorState
        title="Error Loading Favorites"
        message="We couldn't load your favorite Hospital. Please try again."
        onRetry={refetch}
      />
    )
  }

  const ListEmptyComponent = <Empty {...FAVORITE_EMPTY[FAVORITE_TYPES.HOSPITAL]} />

  return (
    <>
      {isLoading && <HospitalListSkeleton paddingHorizontal={24} count={3} />}
      <HospitalList
        data={hospitals}
        ItemSeparatorComponent={ItemSeparatorComponent}
        keyExtractor={keyExtractor}
        estimatedItemSize={height}
        ListEmptyComponent={ListEmptyComponent}
        contentContainerStyle={styles.contentContainerStyle}
      />
    </>
  )
}

export default HospitalFavorite

const styles = StyleSheet.create({
  contentContainerStyle: {
    paddingHorizontal: 24,
    paddingBottom: 8,
  },
})
