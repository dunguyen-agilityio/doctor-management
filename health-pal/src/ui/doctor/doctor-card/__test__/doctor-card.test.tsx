import { useFavoritesStore } from '@/stores/favorite'
import { fireEvent, render, screen } from '@utils-test'

import { ROUTES } from '@/constants'

import { useAddFavorite } from '@/hooks/use-add-favorite'
import * as MockModal from '@/hooks/use-modal'
import { useRemoveFavorite } from '@/hooks/use-remove-favorite'

import { ModalRef } from '@/types/modal'

import DoctorCard from '..'

jest.mock('@/hooks/use-remove-favorite')
jest.mock('@/hooks/use-add-favorite')
jest.mock('@/stores/favorite')

describe('DoctorCard', () => {
  const mockAddFavorite = jest.fn()
  const mockRemoveFavorite = jest.fn()
  const mockRef: React.RefObject<ModalRef> = {
    current: {
      open: jest.fn(),
      close: jest.fn(),
    },
  }
  const defaultProps = {
    id: 1,
    documentId: 'doc1',
    avatar: 'https://example.com/avatar.jpg',
    name: 'Dr. John Doe',
    specialty: 'Cardiology',
    address: '123 Main St, City',
    rating: 4.5,
    reviewCounter: 100,
    showReview: true,
    actionable: true,
  }

  beforeEach(() => {
    jest.clearAllMocks()
    ;(useFavoritesStore as unknown as jest.Mock).mockReturnValue(null)
    ;(useAddFavorite as jest.Mock).mockReturnValue({
      mutate: mockAddFavorite,
      isPending: false,
    })
    ;(useRemoveFavorite as jest.Mock).mockReturnValue({
      mutate: mockRemoveFavorite,
      isPending: false,
    })
    // Mock useModal
    jest.spyOn(MockModal, 'useModal').mockImplementation((ref) => {
      if (ref && ref !== null && 'current' in ref) {
        ref.current = mockRef.current
      }
      return [true, jest.fn()]
    })
  })

  afterEach(() => {
    jest.restoreAllMocks()
  })

  it('renders card content with reviews', () => {
    render(<DoctorCard {...defaultProps} />)

    expect(screen.getByLabelText('Image of Dr. John Doe')).toBeTruthy()
    expect(screen.getByText('Dr. John Doe')).toBeTruthy()
    expect(screen.getByText('Cardiology')).toBeTruthy()
    expect(screen.getByTestId('address-text')).toHaveTextContent('123 Main St, City')
    expect(screen.getByText('100 Reviews')).toBeTruthy()
    expect(screen.getByLabelText('Add Dr. John Doe to favorites')).toBeTruthy()
  })

  it('hides reviews when showReview is false', () => {
    render(<DoctorCard {...defaultProps} showReview={false} />)

    expect(screen.queryByText('100 Reviews')).not.toBeTruthy()
    expect(screen.queryByTestId('mock-stars')).not.toBeTruthy()
    expect(screen.queryByLabelText('Add Dr. John Doe to favorites')).not.toBeTruthy()
  })

  it('renders as non-actionable card without Link', () => {
    render(<DoctorCard {...defaultProps} actionable={false} />)

    expect(screen.queryByTestId('doctor-link')).not.toBeTruthy()
    expect(screen.getByLabelText('Image of Dr. John Doe')).toBeTruthy()
  })

  it('renders Link when actionable is true', () => {
    render(<DoctorCard {...defaultProps} />)

    const link = screen.getByTestId('doctor-link')
    expect(link).toHaveProp('href', ROUTES.DOCTOR.replace('[id]', 'doc1'))
    expect(link).toHaveProp('accessibilityLabel', 'View doctors at Dr. John Doe')
  })

  it('adds favorite when favorite button is pressed and no favoriteId', () => {
    render(<DoctorCard {...defaultProps} />)

    const favoriteButton = screen.getByLabelText('Add Dr. John Doe to favorites')
    fireEvent.press(favoriteButton)

    expect(mockAddFavorite).toHaveBeenCalledWith(1)
    expect(mockRef.current?.open).not.toHaveBeenCalled()
  })

  it('opens confirmation modal when favorite button is pressed with favoriteId', () => {
    ;(useFavoritesStore as unknown as jest.Mock).mockReturnValue('fav1')
    render(<DoctorCard {...defaultProps} />)

    const favoriteButton = screen.getByLabelText('Remove Dr. John Doe from favorites')
    fireEvent.press(favoriteButton)

    expect(mockRef.current?.open).toHaveBeenCalled()
    expect(mockAddFavorite).not.toHaveBeenCalled()
  })

  it('removes favorite when confirmation modal is confirmed', () => {
    ;(useFavoritesStore as unknown as jest.Mock).mockReturnValue('fav1')
    render(<DoctorCard {...defaultProps} />)

    const favoriteButton = screen.getByLabelText('Remove Dr. John Doe from favorites')
    fireEvent.press(favoriteButton)

    const removeButton = screen.getByTestId('remove-favorite-button')
    fireEvent.press(removeButton)

    expect(mockRef.current?.close).toHaveBeenCalled()
    expect(mockRemoveFavorite).toHaveBeenCalledWith('fav1')
  })

  it('has correct accessibility attributes', () => {
    render(<DoctorCard {...defaultProps} />)

    const image = screen.getByLabelText('Image of Dr. John Doe')
    const favoriteButton = screen.getByLabelText('Add Dr. John Doe to favorites')
    const link = screen.getByTestId('doctor-link')

    expect(image).toHaveProp('aria-label', 'Image of Dr. John Doe')
    expect(image).toHaveProp('role', 'img')
    expect(favoriteButton).toHaveProp('aria-label', 'Add Dr. John Doe to favorites')
    expect(favoriteButton).toHaveProp('role', 'button')
    expect(link).toHaveProp('accessibilityLabel', 'View doctors at Dr. John Doe')
    expect(link).toHaveProp('accessibilityHint', 'Shows this doctor in the doctor list')
    expect(link).toHaveProp('role', 'link')
  })

  it('disables card during add/remove favorite operations', () => {
    ;(useAddFavorite as jest.Mock).mockReturnValue({ mutate: mockAddFavorite, isPending: true })
    render(<DoctorCard {...defaultProps} />)

    const card = screen.getByTestId('doctor-link')
    expect(card).toHaveProp('disabled', true)
  })

  it('matches snapshot', () => {
    const { toJSON } = render(<DoctorCard {...defaultProps} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
