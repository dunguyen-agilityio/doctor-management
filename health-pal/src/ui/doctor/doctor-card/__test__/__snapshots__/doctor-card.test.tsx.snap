// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoctorCard matches snapshot 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <Text
    accessibilityHint="Shows this doctor in the doctor list"
    accessibilityLabel="View doctors at Dr. John Doe"
    disabled={false}
    href="/doctors/details/doc1"
    onPress={[Function]}
    role="link"
    testID="doctor-link"
  >
    <View
      style={
        {
          "backgroundColor": "hsla(0, 0%, 100%, 1)",
          "borderBottomColor": "#f3f4f6",
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "borderBottomWidth": 0.5,
          "borderLeftColor": "#f3f4f6",
          "borderLeftWidth": 0.5,
          "borderRightColor": "#f3f4f6",
          "borderRightWidth": 0.5,
          "borderStyle": "solid",
          "borderTopColor": "#f3f4f6",
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "gap": 12,
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 0,
          "paddingTop": 12,
          "position": "relative",
          "shadowColor": "rgb(0,0,0)",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.0392156862745098,
          "shadowRadius": 4,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "transparent",
            "flexDirection": "column",
            "marginBottom": "auto",
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "zIndex": 10,
          }
        }
      >
        <View
          style={
            {
              "borderRadius": 12,
              "height": 110,
              "overflow": "hidden",
              "width": 110,
            }
          }
        >
          <ViewManagerAdapter_ExpoImage
            aria-label="Image of Dr. John Doe"
            cachePolicy="disk"
            containerViewRef={"[React.ref]"}
            contentFit="cover"
            contentPosition={
              {
                "left": "50%",
                "top": "50%",
              }
            }
            height="100%"
            nativeViewRef={"[React.ref]"}
            onError={[Function]}
            onLoad={[Function]}
            onLoadEnd={[Function]}
            onLoadStart={[Function]}
            onProgress={[Function]}
            placeholder={
              [
                {
                  "height": 16,
                  "uri": "blurhash:/LDPzx.00%5B93GrBXnp0McR5X9NHn%23",
                  "width": 16,
                },
              ]
            }
            role="img"
            source={
              [
                {
                  "uri": "https://example.com/avatar.jpg",
                },
              ]
            }
            style={
              {
                "height": "100%",
                "width": "100%",
              }
            }
            tabIndex={0}
            testID="cloudinary-image"
            transition={null}
            width="100%"
          />
          <View
            style={
              {
                "flexDirection": "column",
                "left": "50%",
                "position": "absolute",
                "top": "50%",
                "transform": [
                  {
                    "translateX": -12,
                  },
                  {
                    "translateY": -12,
                  },
                ],
              }
            }
          >
            <ActivityIndicator
              color="#1c64f2"
              size="small"
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "marginBottom": 0,
            "marginTop": 0,
            "paddingBottom": 0,
            "paddingTop": 0,
            "zIndex": 5,
          }
        }
      >
        <View
          aria-label="Add Dr. John Doe to favorites"
          focusVisibleStyle={{}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          role="button"
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomColor": "transparent",
              "borderBottomLeftRadius": 24,
              "borderBottomRightRadius": 24,
              "borderBottomWidth": 1,
              "borderLeftColor": "transparent",
              "borderLeftWidth": 1,
              "borderRightColor": "transparent",
              "borderRightWidth": 1,
              "borderStyle": "solid",
              "borderTopColor": "transparent",
              "borderTopLeftRadius": 24,
              "borderTopRightRadius": 24,
              "borderTopWidth": 1,
              "flexDirection": "row",
              "flexWrap": "nowrap",
              "fontWeight": "500",
              "height": 24,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "position": "absolute",
              "right": 6,
              "top": 0,
              "width": 24,
              "zIndex": 1000,
            }
          }
          tabIndex={0}
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={15}
            bbWidth={15}
            fill="none"
            focusable={false}
            height={15}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 15,
                  "width": 15,
                },
              ]
            }
            vbHeight={18}
            vbWidth={20}
            width={15}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M14.4404 0.599998C17.2317 0.600231 19.4999 2.87386 19.5 5.68984C19.5 6.68349 19.3616 7.6104 19.1162 8.47793L19.0039 8.84609L19.0029 8.84902C18.2487 11.2357 16.7033 13.1595 15.0352 14.5932C13.5737 15.8492 12.0403 16.7106 10.915 17.1664L10.459 17.3363L10.4531 17.3383C10.3532 17.3735 10.189 17.3998 10 17.3998C9.85804 17.3998 9.73007 17.385 9.63281 17.3627L9.54688 17.3383L9.54102 17.3363L9.08496 17.1664C7.9597 16.7106 6.42634 15.8492 4.96484 14.5932C3.40096 13.2491 1.94481 11.4746 1.14746 9.2914L0.99707 8.84902L0.996094 8.84609L0.883789 8.47793C0.63838 7.6104 0.5 6.68349 0.5 5.68984C0.500084 2.87386 2.76829 0.600232 5.55957 0.599998C7.20459 0.599998 8.67953 1.39959 9.59961 2.6293L10 3.16445L10.4004 2.6293C11.3205 1.39959 12.7954 0.599998 14.4404 0.599998Z"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                  ]
                }
                stroke={
                  {
                    "payload": 4280035898,
                    "type": 0,
                  }
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View
          style={[]}
        >
          <View
            style={[]}
          >
            <Text
              style={
                {
                  "color": "#1c2a3a",
                  "fontFamily": "Inter_700Bold",
                  "fontSize": 16,
                  "lineHeight": 24,
                }
              }
              suppressHighlighting={true}
            >
              Dr. John Doe
            </Text>
          </View>
          <View
            style={[]}
          />
          <View
            style={{}}
          >
            <Text
              style={
                {
                  "color": "#1c2a3a",
                  "fontFamily": "Inter_600SemiBold",
                  "fontSize": 14,
                  "lineHeight": 21,
                }
              }
              suppressHighlighting={true}
            >
              Cardiology
            </Text>
            <View
              style={[]}
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="14"
                bbWidth="14"
                fill="none"
                focusable={false}
                height="14"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 14,
                      "width": 14,
                    },
                  ]
                }
                vbHeight={14}
                vbWidth={14}
                width="14"
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M7.00018 7.83421C8.00533 7.83421 8.82018 7.01937 8.82018 6.01421C8.82018 5.00906 8.00533 4.19421 7.00018 4.19421C5.99502 4.19421 5.18018 5.00906 5.18018 6.01421C5.18018 7.01937 5.99502 7.83421 7.00018 7.83421Z"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    propList={
                      [
                        "stroke",
                      ]
                    }
                    stroke={
                      {
                        "payload": 4285231744,
                        "type": 0,
                      }
                    }
                  />
                  <RNSVGPath
                    d="M2.11182 4.95246C3.26099 -0.099206 10.7452 -0.0933725 11.8885 4.95829C12.5593 7.92163 10.716 10.43 9.10016 11.9816C7.92766 13.1133 6.07266 13.1133 4.89432 11.9816C3.28432 10.43 1.44099 7.91579 2.11182 4.95246Z"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    propList={
                      [
                        "stroke",
                      ]
                    }
                    stroke={
                      {
                        "payload": 4285231744,
                        "type": 0,
                      }
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                numberOfLines={1}
                style={
                  {
                    "color": "#1c2a3a",
                    "fontFamily": "Inter_400Regular",
                    "fontSize": 14,
                    "lineHeight": 21,
                    "width": 200,
                  }
                }
                suppressHighlighting={true}
                testID="address-text"
              >
                123 Main St, City
              </Text>
            </View>
          </View>
          <View
            style={[]}
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row-reverse",
                  "gap": 8,
                }
              }
              testID="stars"
            >
              <Text
                style={
                  {
                    "color": "#1c2a3a",
                    "fontFamily": "Inter_600SemiBold",
                    "fontSize": 12,
                    "lineHeight": 18,
                  }
                }
                suppressHighlighting={true}
                testID="stars-text"
              >
                4.5
              </Text>
              <View
                style={[]}
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={15}
                  bbWidth={15}
                  fill="none"
                  focusable={false}
                  height={15}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": 15,
                        "width": 15,
                      },
                    ]
                  }
                  testID="star-0"
                  vbHeight={21}
                  vbWidth={21}
                  width={15}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      clipRule={0}
                      d="M10.6136 0.818184C9.9994 0.849607 9.39008 1.33065 8.91935 2.25989L7.1322 5.83418C6.89138 6.3285 6.24497 6.81014 5.69995 6.89887L2.46787 7.43121C0.401877 7.7861 -0.0797662 9.28173 1.40319 10.7647L3.92548 13.287C4.34374 13.7052 4.58457 14.5291 4.47049 15.1375L3.74803 18.2555C3.17766 20.7144 4.49584 21.665 6.66324 20.3849L9.69252 18.585C9.94502 18.4324 10.2764 18.3504 10.6136 18.3392V0.818184Z"
                      fill={
                        {
                          "payload": 4294956800,
                          "type": 0,
                        }
                      }
                      fillRule={0}
                      propList={
                        [
                          "fill",
                          "fillRule",
                        ]
                      }
                    />
                    <RNSVGPath
                      clipRule={0}
                      d="M10.6133 18.3409C11.0039 18.328 11.4023 18.4099 11.6948 18.5868L14.7241 20.3866C16.8915 21.6794 18.2096 20.7161 17.6393 18.2572L16.9168 15.1392C16.7774 14.5435 17.0182 13.7196 17.4365 13.3014L19.9588 10.7791C21.4417 9.28345 20.9601 7.78782 18.8941 7.4456L15.662 6.91326C15.117 6.81186 14.4706 6.34289 14.2297 5.84857L12.4426 2.27428C11.9374 1.27039 11.2725 0.786183 10.6133 0.819899V18.3409Z"
                      fill={
                        {
                          "payload": 4294956800,
                          "type": 0,
                        }
                      }
                      fillRule={0}
                      propList={
                        [
                          "fill",
                          "fillRule",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
            <View
              style={[]}
            />
            <Text
              style={
                {
                  "color": "#1c2a3a",
                  "fontFamily": "Inter_400Regular",
                  "fontSize": 12,
                  "lineHeight": 18,
                }
              }
              suppressHighlighting={true}
            >
              100 Reviews
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Text>,
]
`;
